perun-cli/unsetRole

```diff
@@ -2,173 +2,130 @@
-		printMessage("AUDITCONSUMERADMIN role was removed from authorizedGroup $authorizedGroupId", $batch);
-	}
-	#if there is other role, only one complementary object is needed
-} elsif ($role eq "CABINETADMIN") {
-	if (defined($facilityId) || defined($voId) || defined($groupId) || defined($resourceId) || defined($serviceId)) {
-		die "ERROR: no other object needed when setting role $role\n";
-	}
-	if (defined($userId)) {
-		$authzResolverAgent->unsetRole( user => $userId, role => $role );
-		printMessage("CABINETADMIN role was removed from user $userId", $batch);
-	} else {
-		die "ERROR: CABINETADMIN role cannot be unset from group \n";
-	}
-	#if there is other role, only one complementary object is needed
-} else {
-	#facility
-	if (defined($facilityId) && !defined($groupId) && !defined($voId) && !defined($resourceId) && !defined($serviceId)) {
-		my $facilitiesAgent = $agent->getFacilitiesAgent;
-		my $facility = $facilitiesAgent->getFacilityById( id => $facilityId );
-		if (defined($userId)) {
-			$authzResolverAgent->unsetRole( user => $userId, complementaryObject => $facility, role => $role );
-			printMessage("$role role was removed from user $userId and facility $facilityId", $batch);
-		} else {
-			$authzResolverAgent->unsetRole( authorizedGroup => $authorizedGroupId, complementaryObject => $facility, role => $role );
-			printMessage("$role role was removed from authorizedGroup $authorizedGroupId and facility $facilityId", $batch);
-		}
-	#group
-	} elsif (!defined($facilityId) && defined($groupId) && !defined($voId) && !defined($resourceId) && !defined($serviceId)) {
-		my $groupsAgent = $agent->getGroupsAgent;
-		my $group = $groupsAgent->getGroupById( id => $groupId );
-		if (defined($userId)) {
-			$authzResolverAgent->unsetRole( user => $userId, complementaryObject => $group, role => $role );
-			printMessage("$role role was removed from user $userId and group $groupId", $batch);
-		} else {
-			$authzResolverAgent->unsetRole( authorizedGroup => $authorizedGroupId, complementaryObject => $group, role => $role );
-			printMessage("$role role was removed from authorizedGroup $authorizedGroupId and group $groupId", $batch);
+# # Complementary object must be set
+# unless (defined($compObjectName)) { die "ERROR: type of complementary object is required\n"; }
+#
+# # Id of comp object must be set
+# unless (defined($objectId)) { die "ERROR: id of complementary object is required\n"; }
+
+my $agent = Perun::Agent->new();
+my $authzResolverAgent = $agent->getAuthzResolverAgent();
+my @rules = $authzResolverAgent->getAllRolesManagementRules();
+
+foreach (@rules) {
+	my $rule = $_;
+	if ($rule->getRoleName() eq $role) {
+		my $primObject = $rule->getPrimaryObject();
+		if (!defined($primObject)) {
+			if (defined($compObjectName)) {
+				die "ERROR: role $role doesn't have a complementary object\n";
+			}
+		} elsif (!defined($compObjectName) || !defined($objectId) || $primObject ne $compObjectName) {
```

Comment:
a nitpick - would be nice to have the complementary object check case insensitive